src/openforms/contrib/brk/validators.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from contextlib import contextmanager
+from typing import Iterator, TypedDict
+
+from django.core.exceptions import ValidationError
+from django.utils.deconstruct import deconstructible
+from django.utils.translation import gettext_lazy as _
+
+from glom import glom
+from requests import RequestException
+from rest_framework import serializers
+from typing_extensions import NotRequired
+
+from openforms.authentication.constants import AuthAttribute
+from openforms.submissions.models import Submission
+from openforms.validations.registry import register
+
+from .client import NoServiceConfigured, SearchParams, get_client
+
+logger = logging.getLogger(__name__)
+
+
+@contextmanager
+def suppress_api_errors(error_message: str) -> Iterator[None]:
+    try:
+        yield
+    except RequestException as e:
+        logger.error(
+            "An exception occured when trying to fetch the BRK API", exc_info=e
+        )
+        raise ValidationError(error_message) from e
+
+
+class AddressValue(TypedDict):
+    postcode: str
+    house_number: str
+    house_letter: NotRequired[str]
+    house_number_addition: NotRequired[str]
+
+
+class AddressValueSerializer(serializers.Serializer):
+    postcode = serializers.RegexField(
+        "^[1-9][0-9]{3} ?(?!sa|sd|ss|SA|SD|SS)[a-zA-Z]{2}$",
+    )
+    house_number = serializers.RegexField(
+        r"^\d{1,5}$",
+    )
+    house_letter = serializers.RegexField(
+        "^[a-zA-Z]$", required=False, allow_blank=True
+    )
+    house_number_addition = serializers.RegexField(
+        "^([a-z,A-Z,0-9]){1,4}$",
+        required=False,
+        allow_blank=True,
+    )
+
+    def validate_postcode(self, value: str) -> str:
+        """Normalize the postcode so that it matches the regex from the BRK API."""
+        return value.upper().replace(" ", "")
```

Comment:
I added a test specific to this serializer, as the current brk tests makes direct use of the plugin's `__call__` method.

It could be added to the `openforms.validations.tests` module, but I feel it doesn't belong there as this test module is "generic", it shouldn't test specific plugin functionality.

Reply:
I agree with the sentiment/concerns - however a test or test case in `openforms.validation.tests` decorated with `@tag("gh-3607")` makes the origin of the test clear. It proves that the feature as a whole works as intended (so it's an integration test).

I'm all for proper abstractions and loose coupling, paired with their respective unit tests, but all those bits combined can still lead to a non-working total solution (classic image of sink + pipes being misalinged ;) ).

Probably the middleground here is inside the `contrib.brk` package to add an API/integration test that calls the validate endpoint specifically with this validator. It also fits our architecture/module organization, as the `forms` and `submissions` apps are considered "core" and may be used by any other Open Forms module.